web/src/components/tasks/anytime-action/tax-clearance-certificate/helpers.ts

```diff
@@ -0,0 +1,73 @@
+import { LookupMunicipalityByName } from "@businessnjgovnavigator/shared/municipality";
+import { StateObject } from "@businessnjgovnavigator/shared/states";
+import { TaxClearanceCertificateData } from "@businessnjgovnavigator/shared/taxClearanceCertificate";
+import { Business } from "@businessnjgovnavigator/shared/userData";
+
+export const getInitialData = (
+  business: Business,
+): {
+  requestingAgencyId: string;
+  businessName: string;
+  addressLine1: string;
+  addressLine2: string;
+  addressCity: string;
+  addressState: StateObject | undefined;
+  addressZipCode: string;
+  taxId: string;
+  taxPin: string;
+} => {
+  const requestingAgencyId = business.taxClearanceCertificateData?.requestingAgencyId || "";
+  const businessName =
+    business?.taxClearanceCertificateData?.businessName || business?.profileData.businessName || "";
+  const addressLine1 =
+    business?.taxClearanceCertificateData?.addressLine1 ||
+    business.formationData.formationFormData.addressLine1 ||
+    "";
+  const addressLine2 =
+    business?.taxClearanceCertificateData?.addressLine2 ||
+    business.formationData.formationFormData.addressLine2 ||
+    "";
+  const addressCity =
+    business?.taxClearanceCertificateData?.addressCity ||
+    LookupMunicipalityByName(business?.formationData?.formationFormData?.addressMunicipality?.name)
+      ?.displayName ||
+    business.formationData.formationFormData.addressCity ||
+    "";
+  const addressState =
+    business?.taxClearanceCertificateData?.addressState ||
+    business.formationData?.formationFormData?.addressState ||
+    undefined;
+  const addressZipCode =
+    business?.taxClearanceCertificateData?.addressZipCode ||
+    business.formationData.formationFormData.addressZipCode ||
+    "";
+  const taxId = business?.taxClearanceCertificateData?.taxId || business.profileData.taxId || "";
+  const taxPin = business?.taxClearanceCertificateData?.taxPin || business.profileData.taxPin || "";
+
+  return {
+    requestingAgencyId,
+    businessName,
+    addressLine1,
+    addressLine2,
+    addressCity,
+    addressState,
+    addressZipCode,
+    taxId,
+    taxPin,
+  };
+};
+
+export const getAllFieldsNonEmpty = (
```

Comment:
This function is not correct, and the test we have for the green state, need to be revisited b/c once this goes away, the test fail.

Reply:
Will take a look.